lib/ClientTlsContext.php

```diff
@@ -293,6 +295,39 @@ public function getSecurityLevel(): int {
         return $this->securityLevel;
     }
 
+    /**
+     * Client certificate to use, if key is no present it assumes it is present in the same file as the certificate.
+     *
+     * @return ClientTlsContext Cloned, modified instance.
+     */
+    public function withClientCertificate(string $certificate, string $key = null): self
+    {
+        $clone = clone $this;
+        $clone->clientCert = $certificate;
+        $clone->clientCertKey = $key;
+
+        return $clone;
+    }
```

Comment:
This should use `Certificate` as `ServerTlsContext` does.

https://github.com/amphp/socket/blob/b213627a46786b473bd20ec79e45dc3f0eb1d89f/lib/ServerTlsContext.php#L253-L258

I guess the `Client` can be dropped in the method name, as this is a `ClientTlsContext`.

Reply:
Should be good

Reply:
@joelwurtz It doesn't use `Certificate` yet.